### ISSUE TITLE
`SubMenu` arrow icon does not render without a supporting font

### ISSUE THREAD

**Issue by Zeenobit** [emilk/egui#3529]
**Describe the bug**
Sub menu items do not render their arrow icon if current fonts do not support the "⏵" glyph:
![Screenshot 2023-11-04 182958](https://github.com/emilk/egui/assets/12216742/d7ee0005-064e-45bf-ad06-a45c13996f1d)

Looking at egui code for `SubMenu`, it is indeed using the "⏵" glyph.
```rust
impl SubMenu {
    fn new(parent_state: Arc<RwLock<MenuState>>, text: impl Into<WidgetText>) -> Self {
        let index = parent_state.write().next_entry_index();
        Self {
            button: SubMenuButton::new(text, "⏵", index),
            parent_state,
        }
    }
}
```

**To Reproduce**

1. Replace default fonts with custom fonts
2. Add a nested `ui.menu_button(...)`
3. Observe "?" being displayed instead of "⏵" on the submenu item

**Expected behavior**
The submenu arrow icon should be font-agnostic, like `CollapsingHeader` arrows:

![Screenshot 2023-11-04 184150](https://github.com/emilk/egui/assets/12216742/53020696-e1e0-4ff0-b237-f6357a8f21d2)

**Comment by LoganDark** [comment]
A more general thing that I've been wanting is the ability to pass vector icons into widgets that expect `WidgetText`. Would that solve this issue? Some way to convert icons into Galleys would probably do it.

**Comment by Zeenobit** [comment]
I don't think so, because as the user, we have very limited control over `SubMenu`. As far as I know there is no public API to customize exactly how the sub/menu gets rendered. The usage of "⏵" is hardcoded.

If egui exposes the API to allow the user to override how the menu and its triangle icon is rendered, then your solution might work.

But even then, I don't see why the user would have to explicitly pass in a vector icon in absence of a glyph, when the `CollapsingHeader` widget seems to be rendering a "⏵" just fine without any glyph or vector icon input.

**Comment by LoganDark** [comment]
> I don't think so, because as the user, we have very limited control over `SubMenu`. As far as I know there is no public API to customize exactly how the sub/menu gets rendered. The usage of "⏵" is hardcoded.

That's not what I'm saying. Egui's own code could use that feature internally, but in order for it to do that the feature has to exist.

> But even then, I don't see why the user would have to explicitly pass in a vector icon in absence of a glyph, when the CollapsingHeader widget seems to be rendering a "⏵" just fine without any glyph or vector icon input.

Doesn't it render a triangle manually? Meanwhile this would allow the SubMenu to render an icon inline with text without manually positioning the icon. I think that is what prompted the SubMenu to use a unicode symbol here.

**Comment by Zeenobit** [comment]
Ah I see what you mean now. Yah, that would make sense.
